File: lib/w_web/controllers/cms/post_controller.ex
defmodule WWeb.CMS.PostController do
  use WWeb, :controller

  alias W.CMS
  alias W.CMS.Post

  action_fallback WWeb.FallbackController

  def index(conn, _params) do
    posts = CMS.list_posts()
    render(conn, "index.json", posts: posts)
  end

  def create(conn, %{"post" => post_params}) do
    with {:ok, %Post{} = post} <- CMS.create_post(post_params) do
      conn
      |> put_status(:created)
      |> put_resp_header("location", Routes.cms_post_path(conn, :show, post))
      |> render("show.json", post: post)
    end
  end

  def show(conn, %{"id" => id}) do
    post = CMS.get_post!(id)
    render(conn, "show.json", post: post)
  end

  def update(conn, %{"id" => id, "post" => post_params}) do
    post = CMS.get_post!(id)

    with {:ok, %Post{} = post} <- CMS.update_post(post, post_params) do
      render(conn, "show.json", post: post)
    end
  end

  def delete(conn, %{"id" => id}) do
    post = CMS.get_post!(id)

    with {:ok, %Post{}} <- CMS.delete_post(post) do
      send_resp(conn, :no_content, "")
    end
  end
end

File: wj/webadmin/src/main/java/me/gsmr/webadmin/controller/PostController.java
package me.gsmr.webadmin.controller;

import me.gsmr.common.model.dto.post.PostDto;
import me.gsmr.webadmin.service.PostServiceClient;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/posts")
public class PostController {

    @Autowired
    private PostServiceClient client;

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public List<PostDto> index() {
        return client.indexPost();
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public PostDto create(@RequestBody PostDto postDto) {
        return client.createPost(postDto);
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public PostDto show(@PathVariable long id) {
        return client.showPost(id);
    }

    @PutMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public PostDto update(@PathVariable long id, @RequestBody PostDto postDto) {
        return client.updatePost(id, postDto);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public Object delete(@PathVariable long id) {
        return client.deletePost(id);
    }
}
